lua/cmp/source.lua

```diff
@@ -359,6 +363,9 @@ source.complete = function(self, ctx, callback)
         end
         self.revision = self.revision + 1
         if #self:get_entries(ctx) == 0 then
```

Comment:
Would be better to also cancel this `get_entries` call on context changed.

Reply:
I just tried adding the cancellation the same way you did it, but then completion no longer always gives the correct results.

It's also not needed I think with my implementation, since the `async.throttle` will stop previous invocations already, so there will always be at most one async filter/complete running.

Reply:
I agree, that code should probably be just removed. Or just be a check on `#self.entries` instead.

@hrsh7th what is the rationale behind this?

We're talking about this line: https://github.com/hrsh7th/nvim-cmp/blob/3ac8d6cd29c74ff482d8ea47d45e5081bfc3f5ad/lua/cmp/source.lua#L361

Reply:
Same remark with the code here https://github.com/hrsh7th/nvim-cmp/blob/3ac8d6cd29c74ff482d8ea47d45e5081bfc3f5ad/lua/cmp/core.lua#L329

Reply:
@yioneko I added abort again, but using `async.abort()`, which yields `abort` and the scheduler simply stops the whole async function. Everything works as expected now. So when the source complete context is aborted, it will never finish. The issue I was having before is that check on `0` that restored the previous entries.

See changes here https://github.com/hrsh7th/nvim-cmp/pull/1583/commits/e7872be8a03146012f4d7d0b535beefe3f9f53c5